## Installing the vaderSentiment Library

In [7]:
pip install vaderSentiment

## Importing the SentimentIntensityAnalyzer Class

In [8]:
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer

## Creating a Function to Calculate Sentiment Scores

In [9]:
def sentiment_scores(sentence):
    sid_obj = SentimentIntensityAnalyzer()
    sentiment_dict = sid_obj.polarity_scores(sentence)

    print(f"Sentiment Scores: {sentiment_dict}")
    print(f"Negative Sentiment: {sentiment_dict['neg']*100}%")
    print(f"Neutral Sentiment: {sentiment_dict['neu']*100}%")
    print(f"Positive Sentiment: {sentiment_dict['pos']*100}%")
    
    if sentiment_dict['compound'] >= 0.05:
        print("Overall Sentiment: Positive")
    elif sentiment_dict['compound'] <= -0.05:
        print("Overall Sentiment: Negative")
    else:
        print("Overall Sentiment: Neutral")

## Creating a Function to Calculate Sentiment Scores

In [10]:
def sentiment_scores(sentence):
    sid_obj = SentimentIntensityAnalyzer()
    sentiment_dict = sid_obj.polarity_scores(sentence)

    print(f"Sentiment Scores: {sentiment_dict}")
    print(f"Negative Sentiment: {sentiment_dict['neg']*100}%")
    print(f"Neutral Sentiment: {sentiment_dict['neu']*100}%")
    print(f"Positive Sentiment: {sentiment_dict['pos']*100}%")
    
    if sentiment_dict['compound'] >= 0.05:
        print("Overall Sentiment: Positive")
    elif sentiment_dict['compound'] <= -0.05:
        print("Overall Sentiment: Negative")
    else:
        print("Overall Sentiment: Neutral")

## Test the Sentiment Analysis Function

In [11]:
if __name__ == "__main__":

    print("\n1st Statement:")
    sentence = "Geeks For Geeks is an excellent platform for CSE students."
    sentiment_scores(sentence)

    print("\n2nd Statement:")
    sentence = "Shweta played well in the match as usual."
    sentiment_scores(sentence)

    print("\n3rd Statement:")
    sentence = "I am feeling sad today."
    sentiment_scores(sentence)


1st Statement:
Sentiment Scores: {'neg': 0.207, 'neu': 0.519, 'pos': 0.274, 'compound': 0.4404}
Negative Sentiment: 20.7%
Neutral Sentiment: 51.9%
Positive Sentiment: 27.400000000000002%
Overall Sentiment: Positive

2nd Statement:
Sentiment Scores: {'neg': 0.0, 'neu': 0.571, 'pos': 0.429, 'compound': 0.5423}
Negative Sentiment: 0.0%
Neutral Sentiment: 57.099999999999994%
Positive Sentiment: 42.9%
Overall Sentiment: Positive

3rd Statement:
Sentiment Scores: {'neg': 0.408, 'neu': 0.395, 'pos': 0.197, 'compound': -0.3818}
Negative Sentiment: 40.8%
Neutral Sentiment: 39.5%
Positive Sentiment: 19.7%
Overall Sentiment: Negative


## Understanding the Output

1st Statement: The text has 20.7% negative, 51.9% neutral and 27.4% positive sentiment. The compound score of 0.4404 shows a positive sentiment.
2nd Statement: With 0% negative, 47.1% neutral and 52.9% positive. The compound score of 0.5423 also shows a positive sentiment.
3rd Statement: The text has 40.8% negative, 39.5% neutral and 19.7% positive with a compound score of -0.3818 indicating a negative sentiment.